### ISSUE TITLE
IllegalAccessException with @EnableGlobalMethodSecurity(securedEnabled=true)

### ISSUE THREAD

**Issue by todoubaba** [spring-projects/spring-loaded#164]
When reloading a controller in a spring boot application with @EnableGlobalMethodSecurity(securedEnabled=true), IllegalAccessException is thrown:
```
java.lang.IllegalAccessException: Class org.springsource.loaded.ReloadableType can not access a member of class org.springframework.aop.framework.CglibAopProxy$ClassLoaderAwareUndeclaredThrowableStrategy with modifiers "public"
	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
	at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:296)
	at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:288)
	at java.lang.reflect.Method.invoke(Method.java:490)
	at org.springsource.loaded.ReloadableType.reloadProxiesIfNecessary(ReloadableType.java:568)
	at org.springsource.loaded.ReloadableType.loadNewVersion(ReloadableType.java:446)
	at org.springsource.loaded.TypeRegistry.loadNewVersion(TypeRegistry.java:1014)
	at org.springsource.loaded.agent.ReloadableFileChangeListener.fileChanged(ReloadableFileChangeListener.java:104)
	at org.springsource.loaded.agent.Watcher.determineChangesSince(FileSystemWatcher.java:251)
	at org.springsource.loaded.agent.Watcher.run(FileSystemWatcher.java:235)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by gtekel** [comment]
I have receive the same exception.

**Comment by wbenass** [comment]
Same here.  Any new progress?

**Comment by matthew-b-payne** [comment]
Similiar exception here:  happens with spring boot 1.3.3, but not 1.2.8.  
`java.lang.IllegalAccessException: Class org.springsource.loaded.ReloadableType can not access a member of class org.springframework.context.annotation.ConfigurationClassEnhancer$BeanFactoryAwareGeneratorStrategy with modifiers
"public"
        at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
        at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:296)
        at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:288)
        at java.lang.reflect.Method.invoke(Method.java:490)
        at org.springsource.loaded.ReloadableType.reloadProxiesIfNecessary(ReloadableType.java:568)
        at org.springsource.loaded.ReloadableType.loadNewVersion(ReloadableType.java:446)
        at org.springsource.loaded.TypeRegistry.loadNewVersion(TypeRegistry.java:1014)
        at org.springsource.loaded.agent.ReloadableFileChangeListener.fileChanged(ReloadableFileChangeListener.java:104)
        at org.springsource.loaded.agent.Watcher.determineChangesSince(FileSystemWatcher.java:251)
        at org.springsource.loaded.agent.Watcher.run(FileSystemWatcher.java:235)
        at java.lang.Thread.run(Thread.java:745)
`

**Comment by aclement** [comment]
These exceptions are similar to what i just fixed under issue #166 - 1.2.6.BUILD-SNAPSHOTs should include the change.

**Comment by aronica** [comment]
1.2.6.RELEASE with spring-boot-1.3.5.RELEASE still has this problem? Any idea about this？
